tests/unit/generator/test_generator.py

```diff
@@ -246,6 +249,86 @@ def test_get_filename_with_proto_and_sub():
     ) == 'bar/types/baz/bacon.py'
 
 
+def test_parse_sample_paths(fs):
+    for fpath in [
+            'sample.yaml',
+            'sampledir/sample.yaml',
+            'other_sampledir/sample.yaml',
+    ]:
+        fs.create_file(
+            fpath,
+            contents=dedent(
+                '''
+                ---
+                type: com.google.api.codegen.SampleConfigProto
+                config_schema_version: 1.2.0
+                samples:
+                - service: google.cloud.language.v1.LanguageService
+                '''
+            )
+        )
+
+    opts = options.Options.build(
+        ("gapic_samples=sample.yaml,"
+         "gapic_samples=sampledir/,"
+         "gapic_samples=other_sampledir"))
+
+    expected_configs = (
+        'sample.yaml',
+        'sampledir/sample.yaml',
+        'other_sampledir/sample.yaml',
+    )
+
+    assert opts.sample_configs == expected_configs
+
+
+@mock.patch(
+    'gapic.samplegen.samplegen.generate_sample',
+    return_value=('sample.py', '')
+)
+@mock.patch(
+    'gapic.samplegen.samplegen.generate_manifest',
+    return_value=('manifest.yaml', yaml.Doc([]))
+)
+def test_samplegen_end_to_end(mock_generate_sample, mock_generate_manifest, fs):
```

Comment:
I wouldn't call this "end_to_end", because in some sense it's not: it does not go through protoc. How about `test_samplegen_config_file_to_output_files`, or some variation thereof?

Reply:
Went with `test_samplegen_config_file_to_output_files`.